src/server.rs

```diff
@@ -138,15 +139,15 @@ pub struct SccacheServer<C: CommandCreatorSync> {
 impl<C: CommandCreatorSync> SccacheServer<C> {
     pub fn new(port: u16,
                pool: CpuPool,
+               core: Core,
                storage: Arc<Storage>) -> io::Result<SccacheServer<C>> {
-        let core = Core::new()?;
         let handle = core.handle();
         let addr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), port);
         let listener = TcpListener::bind(&SocketAddr::V4(addr), &handle)?;
 
         // Prepare the service which we'll use to service all incoming TCP
         // connections.
-        let (tx, rx) = mpsc::channel(100);
+        let (tx, rx) = mpsc::channel(1);
```

Comment:
Oh this was actually an error from before, I intended it to be 1 in the first place and set it to 100 in debugging when I thought something was going wrong, I just forgot to change it back until this commit!